compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -470,7 +472,7 @@ object Semantic:
     def widenArg: Contextual[Value] =
       a match
       case _: Ref | _: Fun =>
-        val hasError = Reporter.hasErrors { a.promote("Argument cannot be promoted to hot") }
+        val hasError = Reporter.hasErrors { a.promote("Argument cannot be promoted to transitively initialized (Hot)") }
```

Comment:
@liufengyun Should we consider saying something like "proven" instead of "promoted"? e.g. "Argument cannot be proven to be transitively initialized"

Reply:
In the documentation, we say "promotion" and have a definition for it. It seems to me "prove" is better than "promoted".  We can see if we can change the definition in the docs as well --- something like "provably hot".